asciimatics/screen.py

```diff
@@ -1914,6 +1914,7 @@ def close(self, restore=True):
                 curses.echo()
                 curses.nocbreak()
                 curses.endwin()
+                signal.signal(signal.SIGWINCH, signal.SIG_DFL)
```

Comment:
it's a bit inefficient, but not a regular occurrence in normal operation, so I think it's OK.  You could try simply disabling the reset inside the signal handler and see if resizes are still handled correctly, but I could live without the optimization.